funsor/torch/distributions.py

```diff
@@ -154,6 +155,19 @@ def _infer_param_domain(cls, name, raw_shape):
     return Real
 
 
+###########################################################
+# Converting distribution funsors to PyTorch distributions
+###########################################################
+
+@to_data.register(Multinomial)  # noqa: F821
+def multinomial_to_data(funsor_dist, name_to_dim=None):
+    probs = to_data(funsor_dist.probs, name_to_dim)
+    total_count = to_data(funsor_dist.total_count, name_to_dim)
+    if isinstance(total_count, numbers.Number) or len(total_count.shape) == 0:
```

Comment:
Do we need to worry about `int(total_count)` thwarting PyTorch tracing? Should we preserve scalar `Tensor`s?

Reply:
I made this change because `torch.distributions.Multinomial` raises an error given a tensor `total_count`, even if it is a scalar tensor, and the generic conversion tests were failing.  It's definitely an issue, but there are already lots of JIT issues with scalars and tuple shapes throughout the codebase (encountered while working on `pyro.contrib.funsor`), and my inclination with this PR and further work on distributions is to avoid special-casing to the greatest extent possible even if that means deferring to odd implementation details in the backends.  I think a better fix in this instance would be to allow scalar tensor `total_count` upstream so this change could be reverted.

This is also a good reminder to add some generic JIT tests for distribution wrappers in a followup PR.